src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py

```diff
@@ -0,0 +1,387 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from msrestazure.azure_exceptions import CloudError
+from azure.cli.core.commands.arm import parse_resource_id, is_valid_resource_id
+import azure.cli.core.azlogging as azlogging
+from azure.cli.core._util import CLIError
+from .custom import set_vm, _compute_client_factory
+logger = azlogging.get_az_logger(__name__)
+
+_DATA_VOLUME_TYPE = 'DATA'
+_STATUS_ENCRYPTED = 'Encrypted'
+
+extension_info = {
+    'Linux': {
+        'publisher': 'Microsoft.Azure.Security',
+        'name': 'AzureDiskEncryptionForLinux',
+        'version': '0.1'
+    },
+    'Windows': {
+        'publisher': 'Microsoft.Azure.Security',
+        'name': 'AzureDiskEncryption',
+        'version': '1.1'
+    }
+}
+
+
+def keyvault_mgmt_client_factory(**_):
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    from azure.mgmt.keyvault import KeyVaultManagementClient
+    return get_mgmt_service_client(KeyVaultManagementClient)
+
+
+def enable(resource_group_name, vm_name,  # pylint: disable=too-many-arguments,too-many-locals, too-many-statements
+           aad_client_id,
+           disk_encryption_keyvault,
+           aad_client_secret=None, aad_client_cert_thumbprint=None,
+           key_encryption_keyvault_id=None,
+           key_encryption_key_url=None,
+           key_encryption_algorithm='RSA-OAEP',
```

Comment:
key_encryption_keyvault_id is really long. Why would "keyvault" (which should accept a name or id) not be sufficient? What is the difference between the disk_encryption_keyvault and the key_encryption_keyvault_id?  
Why not simply `key_uri` as well?